ipaserver/plugins/sudorule.py

```diff
@@ -592,7 +592,25 @@ def pre_callback(self, ldap, dn, found, not_found, *keys, **options):
             raise errors.MutuallyExclusiveError(
                 reason=_("users cannot be added when user category='all'"))
 
-        return add_external_pre_callback('user', ldap, dn, keys, options)
+        dnx = add_external_pre_callback('user', ldap, dn, keys, options)
+        if 'trusted_objects' in options:
+            trusted_objects = options.pop('trusted_objects')
+            found_members = found.get('memberuser', {})
+            found_users = found_members.get('user', [])
+            filtered_users = []
+            for user in found_users:
+                for obj in trusted_objects:
+                    m = obj
+                    if obj.startswith('%:'):
+                        m = obj[2:]
+                    if user[0].value != m:
```

Comment:
The comparison is done on strings but since it contains a string representation of a uid it may lead to wrong results (administrator@addomain.test and Administrator@addomain.test would be seen as different although they represent the same object).

Reply:
We can do `str.lower()` for comparison, but this would break SSSD's https://github.com/SSSD/sssd/pull/5251 once it is implemented.

Reply:
In my testing user[0].value = 'Administrator@ADROOT', m = 'administrator@adroot' so it was always including a not found. Once I added lower() it resolved it and I stopped getting a not found failure reported (even though the user was also added as a member).